### PR TITLE
Data Objects: Improve IdInventory, IdCodec extensibility

### PR DESCRIPTION
**Pull request by paolobazzi**
- IdCodec: Improve extensibility
- IdCodec: Handle null/empty string as null-id
- IdCodec: Add abstract test and default test implementation
- IdInventory: Add method to unregister type mapper
- Add unit tests

322156

### CODE REVIEW

File: org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java [hunk 1]
```diff
@@ -235,4 +247,16 @@ protected Object[] parseComponents(Class<? extends IId> idClass, String[] rawCom
...
   }
+
+  /**
+   * Unregister type mapping between a string representation and the given {@code rawType}.
+   * <p>
+   * Note: The access to the type mapping data structure is not synchronized and therefore not thread safe. Use this
+   * method to set up the {@link IdCodec} instance directly after platform start and not to change the {@link IdCodec}
+   * behavior dynamically at runtime.
+   */
+  protected void unregisterRawTypeMapper(Class<?> rawType) {
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java]: Position directly below the register method further above

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/id/IdCodecTest.java [hunk 2]
```diff
@@ -10,520 +10,15 @@
...
 
-    public static FixtureCustomComparableRawDataId of(CustomComparableRawDataType date) {
-      if (date == null) {
-        return null;
-      }
-      return new FixtureCustomComparableRawDataId(date);
-    }
+  @Override
+  protected IdCodec getCodec() {
+    return BEANS.get(IdCodec.class);
```
Comment (stephan-merkli) [inline org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/id/IdCodecTest.java]: Always get the instance again or use member variable instead?

File: org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java (hunk 1 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java]: Done

File: org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/id/IdCodecTest.java (hunk 2 above)
Comment (paolobazzi) [reply org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/id/IdCodecTest.java]: Done

**Comment by paolobazzi** [comment]
Please run tests

**Pull request merged**